cibuildwheel/oci_container.py

```diff
@@ -75,6 +96,32 @@ def options_summary(self) -> str | dict[str, str]:
 DEFAULT_ENGINE = OCIContainerEngineConfig("docker")
 
 
+def _check_engine_version(engine: OCIContainerEngineConfig) -> None:
+    try:
+        version_string = call(engine.name, "version", "-f", "{{json .}}", capture_stdout=True)
+        version_info = json.loads(version_string.strip())
+        if engine.name == "docker":
+            # --platform support was introduced in 1.32 as experimental
+            # docker cp, as used by cibuildwheel, has been fixed in v24 => API 1.43, https://github.com/moby/moby/issues/38995
+            client_api_version = Version(version_info["Client"]["ApiVersion"])
+            engine_api_version = Version(version_info["Server"]["ApiVersion"])
+            version_supported = min(client_api_version, engine_api_version) >= Version("1.43")
```

Comment:
I wonder about the use of package.version.Version for parsing version strings from outside the python ecosystem. Browsing the [Docker release notes](https://docs.docker.com/engine/release-notes/18.05/), some of the versions look like '[18.05.0-ce](https://docs.docker.com/engine/release-notes/18.05/#18050-ce)', which wont parse with Version:

```python
>>> Version('17.09.1-ce')
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Users/joerick/Projects/cibuildwheel/env/lib/python3.11/site-packages/packaging/version.py", line 200, in __init__
    raise InvalidVersion(f"Invalid version: '{version}'")
packaging.version.InvalidVersion: Invalid version: '17.09.1-ce'
```

I can't test the exact string that will show up there but I think we'd need to be more flexible with the string to ensure we don't fail to parse it.

Reply:
I think this issue only applies to the "Version" field which can indeed be like `17.09.1-ce`. It should not apply to "APIVersion".

Reply:
ah, yes, `x.y` shouldn't pose an issue.